### PR TITLE
interstitial forking rename dialog

### LINKED ISSUE #251: Interstitial forking rename dialog
This is the desired layout and elements:

image

This is the desired visual style (established in codebase):

image

### PR DESCRIPTION
**Pull request by stushurik**
Closes datavis-tech/vizhub-issue-tracker/issues/251

### CODE REVIEW

**Comment by curran** [comment]
@stushurik When I run it, there are errors:

image

Please fix these and I will review again. Thanks!

**Comment by stushurik** [comment]
@curran fixed

File: packages/useCases/src/interactors/forkVisualization.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ export class ForkVisualization {
       this.visualizationGateway.createVisualization({
         owner,
         id: generateId(),
-        title: visualization.info.title,
+        title: forkSettings.forkTitle || visualization.info.title,
```
Comment (curran) [inline packages/useCases/src/interactors/forkVisualization.js:50]: This is good, but in the forked viz, the `<title>` tag in `index.html` should also be updated.

**Review by curran — CHANGES_REQUESTED** [review]
It's great so far, but one issue is that in the forked viz, the `<title>` tag content is not updated to the new name.

image

Please add some logic that makes that change to `index.html` in the forked viz.

Here's a good entry point to study what's in the codebase so far around the title in HTML: https://github.com/curran/vizhub/blob/master/packages/neoFrontend/src/pages/VizPage/RunContext/updateTitleIfNeeded.js

I think this regex will be useful: https://github.com/curran/vizhub/blob/master/packages/presenters/src/accessors.js#L56

```
export const extractTitle = (html) => {
  if (html) {
    const titleMatch = html.match(/<title>(.*?)<\/title>/i);
    if (titleMatch) {
      return titleMatch[1];
    }
  }
  return 'Untitled';
};
```

While you're in this part of the codebase, maybe also consider this one https://github.com/datavis-tech/vizhub-issue-tracker/issues/619 . Ideally the title could contain newlines, which may be a simple modification of the regex (or might be more complex, not sure).

Another minor thing - if the user does not make any changes to the title, clicking on "Fork" does nothing. This behavior is confusing. I suggest to allow the user to make for fork and leave the default provided title, rather than stopping them from forking until they have edited it.

Thanks!

**Comment by curran** [comment]
I made one commit here by the way, to fix the unfinished merge conflict resolution. Also ran Prettier. So be sure to pull the latest!

File: packages/useCases/src/interactors/forkVisualization.js [hunk 2]
```diff
@@ -21,6 +22,22 @@ export class ForkVisualization {
...
     const nowTimestamp = timestamp();
 
+    const indexHtmlIndex = getFileIndex(
+      visualization.content.files,
+      'index.html'
+    );
+    const indexHtml = visualization.content.files[indexHtmlIndex];
+    const updatedIndexHtml = {
+      ...indexHtml,
+      text: indexHtml.text.replace(
```
Comment (curran) [inline packages/useCases/src/interactors/forkVisualization.js:32]: Beautiful! Nice simple solution.

**Review by curran — APPROVED** [review]
Looks fantastic! Thanks for doing this.

**Pull request merged**